packages/frontend/src/components/Swap/SwapAndSuccessContainer.js

```diff
@@ -0,0 +1,156 @@
+import React, { useEffect, useState } from 'react';
+import styled from 'styled-components';
+import Container from '../common/styled/Container.css';
+import { currentToken } from './helpers';
+import { useDispatch } from 'react-redux';
+import { handleSwapBycontractName } from '../../redux/slices/swap';
+import SwapPage from './views/SwapPage';
+import Success from './views/Success';
+
+export const VIEWS_SWAP = {
+    MAIN: 'main',
+    SUCCESS: 'success'
+};
+
+const StyledContainer = styled(Container)`
+    position: relative;
+
+    h1 {
+        text-align: center;
+        margin-bottom: 30px;
+    }
+    .text {
+        margin-bottom: 11px;
+    }
+
+    .iconSwap {
+        margin: 0 auto;
+        width: fit-content;
+        margin-bottom: 30px;
+
+        svg {
+            transform: rotate(90deg);
+            cursor: pointer;
+        }
+    }
+
+    .buttons-bottom-buttons {
+        margin-top: 30px;
+
+        > button {
+            display: block;
+            width: 100%;
+        }
+
+        .link {
+            display: block !important;
+            margin: 20px auto !important;
+        }
+    }
+
+    .text_info_success {
+        width: fit-content;
+        font-style: normal;
+        font-weight: 700;
+        font-size: 20px;
+        line-height: 28px;
+        text-align: center;
+        color: #252729;
+        margin: 0 auto;
+    }
+`;
+
+
+const SwapAndSuccessContainer = ({
+    fungibleTokensList,
+    accountId,
+    miltiplier,
+    swapContractValue,
+    activeView,
+    isLoading,
+    handleSwapToken,
```

Comment:
I am not sure why this needs to be passed in two levels here, this functionality seems specific to `SwapPage` component why not create and use a memoized callback there?